pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -727,60 +727,126 @@ async def _map_user_message(self, message: ModelRequest) -> AsyncIterable[chat.C
+        if item.is_audio:
+            assert item.format in ('wav', 'mp3')
+            audio = InputAudio(data=base64_encoded, format=item.format)
+            return [ChatCompletionContentPartInputAudioParam(input_audio=audio, type='input_audio')]
+        if item.is_document:
+            return [
+                File(
+                    file=FileFile(
+                        file_data=f'data:{item.media_type};base64,{base64_encoded}',
+                        filename=f'filename.{item.format}',
+                    ),
+                    type='file',
+                )
+            ]
+        return None
+
+    @staticmethod
+    async def _handle_audio_url(item: object) -> list[ChatCompletionContentPartParam] | None:
+        if not isinstance(item, AudioUrl):
+            return None
+        downloaded_item = await download_item(item, data_format='base64', type_format='extension')
+        assert downloaded_item['data_type'] in ('wav', 'mp3'), (
+            f'Unsupported audio format: {downloaded_item["data_type"]}'
+        )
+        audio = InputAudio(data=downloaded_item['data'], format=downloaded_item['data_type'])
+        return [ChatCompletionContentPartInputAudioParam(input_audio=audio, type='input_audio')]
+
+    @staticmethod
+    async def _handle_document_url(item: object) -> list[ChatCompletionContentPartParam] | None:
+        if not isinstance(item, DocumentUrl):
+            return None
+        if OpenAIChatModel._is_text_like_media_type(item.media_type):
+            downloaded_text = await download_item(item, data_format='text', type_format='extension')
+            inline = OpenAIChatModel._inline_file_block(item.media_type, downloaded_text['data'], file_url=item.url)
+            return [ChatCompletionContentPartTextParam(text=inline, type='text')]
+        downloaded_item = await download_item(item, data_format='base64_uri', type_format='extension')
+        return [
+            File(
+                file=FileFile(
+                    file_data=downloaded_item['data'],
+                    filename=f'filename.{downloaded_item["data_type"]}',
+                ),
+                type='file',
+            )
+        ]
+
+    @staticmethod
+    def _is_text_like_media_type(media_type: str) -> bool:
+        return (
+            media_type.startswith('text/')
+            or media_type == 'application/json'
+            or media_type.endswith('+json')
+            or media_type == 'application/xml'
+            or media_type.endswith('+xml')
+            or media_type == 'text/csv'
+            or media_type == 'text/markdown'
+            or media_type in ('application/x-yaml', 'application/yaml', 'text/yaml')
+        )
+
+    @staticmethod
+    def _inline_file_block(media_type: str, text: str, file_url: str | None = None) -> str:
+        if file_url is not None:
+            return f'-----BEGIN FILE url="{file_url}" type="{media_type}"-----\n{text}\n-----END FILE-----'
```

Comment:
Could we include the `identifier` field instead of the URL, which already exists for this purpose? And document in its docstring that it's used in this situation.

Also while you're at it, we can remove "generating one from the url if not explicitly set" from the docstring, looks like it made it in there by accident